packages/ecc-utils-design/src/components/details/details.ts

```diff
@@ -193,100 +109,172 @@ export default class EccUtilsDesignDetails extends LitElement {
     this.requestUpdate();
   }
 
-  private _renderText(child: Children): TemplateResult {
-    const content = html`<div class="value">${child.value}</div>`;
-    return this._renderContainer(child, content);
-  }
-
-  private _renderLongText(child: Children): TemplateResult {
-    const content = html`<div id="${child.key}" class="value large-text">
-      ${child.value}
-    </div>`;
-    return this._renderContainer(child, content, true);
-  }
-
-  private _renderObjectItem(
-    [key, value]: [string, string],
-    child: Children
+  private _renderData(
+    data: string,
+    label: string,
+    copy = false
   ): TemplateResult {
+    if (data === null || data === undefined) return html``;
     return html`
-      <div class="object-item">
-        <div class="object-key">${key}:</div>
-        <div id="${child.key}" class="object-value">${value}</div>
-        ${child.textOptions?.copy &&
-        html`<sl-copy-button
-          class="copy-button"
-          from=${child.key}
-        ></sl-copy-button>`}
+      <div class="container">
+        <div class="label">
+          <span>${label}</span>
+          ${copy
+            ? html`
+                <sl-copy-button
+                  value=${data}
+                  copy-label="Click to copy ${label}"
+                  success-label="${label} copied"
+                  error-label="Error"
+                ></sl-copy-button>
+              `
+            : html``}
+        </div>
+        <div class="value">${data}</div>
       </div>
     `;
   }
 
-  private _renderObject(child: Children): TemplateResult {
+  private _renderArray(
+    data: Array<any>,
+    label: string,
+    copy = false
+  ): TemplateResult {
+    if (data === null || data === undefined || data.length === 0) return html``;
     return html`
-      <div class="details-container details-container-object">
-        <div class="label">${child.label}</div>
-        <div class="object-value-container">
-          ${Object.entries(child.value).map((entry) =>
-            this._renderObjectItem(entry, child)
-          )}
+      <div class="container">
+        <div class="label">
+          <span>${label}</span>
+          ${copy
+            ? html`
+                <sl-copy-button
+                  value=${JSON.stringify(data)}
+                  copy-label="Click to copy strigified JSON of ${label}"
+                  success-label="${label} copied"
+                  error-label="Error"
+                ></sl-copy-button>
+              `
+            : html``}
+        </div>
+        <div class="value">
+          ${data.map((value, index) => {
```

Comment:
you're using very similar logic in 2 other places, the `_renderObject` method and the `_renderField` method. Perhaps you could convert it to a function. Also, is there no option to copy sub-objects or sub-arrays?